FinSearchUnified/Bundle/SearchBundleFindologic/QueryBuilder.php

```diff
@@ -166,4 +167,102 @@ private function getClientIp()
 
         return $ipAddress;
     }
+
+    /**
+     * @return array
+     */
+    public function getParameters()
+    {
+        return $this->parameters;
+    }
+
+    /**
+     * @param string $query
+     */
+    public function addQuery($query)
+    {
+        $this->parameters['query'] = urldecode($query);
+    }
+
+    /**
+     * @param float $min
+     * @param float $max
+     */
+    public function addPrice($min, $max)
+    {
+        $this->addParameter('price', ['min' => urldecode($min), 'max' => urldecode($max)]);
+    }
+
+    /**
+     * @param string $key
+     * @param mixed $value
+     */
+    public function addFilter($key, $value)
+    {
+        $this->addParameter($key, urldecode($value));
+    }
+
+    /**
+     * @param array $categories
+     */
+    public function addCategories(array $categories)
+    {
+        $this->addParameter('cat', $categories);
+    }
+
+    /**
+     * @param string $order
+     */
+    public function addOrder($order)
+    {
+        $this->parameters['order'] = $order;
+    }
+
+    /**
+     * @param int $firstResult
+     */
+    public function setFirstResult($firstResult)
+    {
+        $this->parameters['first'] = $firstResult;
+    }
+
+    /**
+     * @param int $maxResults
+     */
+    public function setMaxResults($maxResults)
+    {
+        $this->parameters['count'] = $maxResults;
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return mixed|null
+     */
+    private function getParameter($key)
+    {
+        if (!isset($this->parameters[self::PARAMETER_KEY_ATTRIB]) ||
+            !isset($this->parameters[self::PARAMETER_KEY_ATTRIB][$key])) {
+            return null;
+        }
+
+        return $this->parameters[self::PARAMETER_KEY_ATTRIB][$key];
+    }
+
+    /**
+     * @param string $key
+     * @param mixed $value
+     */
+    private function addParameter($key, $value)
+    {
+        if (!isset($this->parameters[self::PARAMETER_KEY_ATTRIB])) {
+            $this->parameters[self::PARAMETER_KEY_ATTRIB] = [];
+        }
+
+        if (!isset($this->parameters[self::PARAMETER_KEY_ATTRIB][$key])) {
+            $this->parameters[self::PARAMETER_KEY_ATTRIB][$key] = [$value];
```

Comment:
Unfortunately you changed this too early. I was about to say that you are right and this doesn't need to be changed.

Reply:
No problem, will revert it again and the test as well.